mod_geoip.c

```diff
@@ -268,47 +267,55 @@ static void geoip_child_init(apr_pool_t * p, server_rec * s)
     geoip_server_config_rec *cfg;
     int i, flags;
 
-    cfg = (geoip_server_config_rec *)
-          ap_get_module_config(s->module_config, &geoip_module);
-
-    if (cfg->gips) {
-        if (cfg->GeoIPFilenames != NULL) {
-            for (i = 0; i < cfg->numGeoIPFiles; i++) {
-                flags =
-                    (cfg->GeoIPFlags2[i] ==
-                     GEOIP_UNKNOWN) ? cfg->GeoIPFlags : cfg->GeoIPFlags2[i];
-                if (flags & (GEOIP_MEMORY_CACHE | GEOIP_MMAP_CACHE)) {
-                    continue;
+    while(s) {
```

Comment:
I am least sure about these loops. I guess I will need to take a closer look at the code. This would be simplified if there was a reliable test case, although I realize that may be difficult for random crashes.